### PR TITLE
Load installed adapter profile for direct commands

### LINKED ISSUE #102: Load the installed adapter profile for direct CLI commands
## Problem

The supported direct `punaro-adapter create`, `send`, and `attachment-notify` commands construct a client from process environment only. The installer writes the validated owner-only client profile and the managed service loads it, but a user invoking the documented direct commands does not get that profile automatically and receives a configuration error before contacting the relay.

## Required outcome

- Define one safe profile-loading contract for direct commands and the managed service.
- Preserve owner-only, non-symlink configuration and credential-file validation.
- Do not print configuration values, credentials, URLs, or message bodies in errors.
- Update operator/client documentation so interactive commands and service execution use the same supported configuration path.

## Acceptance evidence

Add focused CLI tests using an installer-shaped profile: create, send, and attachment-notify reach their transport boundary without ambient exported configuration; malformed or unsafe profiles fail closed; existing explicit environment overrides remain intentional and documented.

### PR DESCRIPTION
**Pull request by rock3r**
## Summary

Fixes #102.

- Load the installed private adapter profile for direct `create`, `send`, and `attachment-notify` commands, using the same validated path as managed services.
- Parse an allowlisted plain-data profile, reject malformed, group-readable, and symlinked files without exposing values, and retain explicit non-empty environment overrides.
- Update service launchers and client/operator documentation so both execution modes follow one supported profile contract.

## Validation

- Focused CLI transport/profile tests (written red first)
- `make ci`

### CODE REVIEW

File: cmd/punaro-adapter/private_file_acl_darwin.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#include <errno.h>
+
+static int punaro_has_extended_acl(int fd) {
+	errno = 0;
+	acl_t acl = acl_get_fd_np(fd, ACL_TYPE_EXTENDED);
+	// APFS reports either ENOATTR or ENOENT for a file with no extended ACL.
+	if (acl == NULL) return (errno == ENOATTR || errno == ENOENT) ? 0 : 1;
+	acl_entry_t entry;
+	int result = acl_get_entry(acl, ACL_FIRST_ENTRY, &entry);
+	while (result == 0) {
```
[the remark this answers is not in the document]
Comment (rock3r) [reply cmd/punaro-adapter/private_file_acl_darwin.go:16]: This has been exercised on macOS by `TestLoadConfigRejectsProfileWithMacOSACL`, which adds `everyone allow read` and passes only because `loadConfig` rejects the profile. On this host, `man acl_get_entry` documents success as `0` (and `-1` otherwise), matching the loop. Changing it to `1` makes ordinary profile tests fail and does not match the platform API, so no code change is warranted.

File: cmd/punaro-adapter/main.go [hunk 2]
```diff
@@ -404,21 +432,72 @@ func loadConfig() (adapterConfig, error) {
...
+	}
+	// #nosec G703 -- this is the fixed installer profile location or an explicit
+	// local operator override; remote data never selects it.
+	if _, err := os.Lstat(path); err != nil {
+		if !explicitPath && errors.Is(err, os.ErrNotExist) {
+			return map[string]string{}, nil
+		}
+		return nil, errors.New("adapter profile is unavailable")
+	}
+	raw, err := readPrivateFile(path, "adapter profile", maxAdapterProfileSize)
```
[the remark this answers is not in the document]
Comment (rock3r) [reply cmd/punaro-adapter/main.go:478]: Addressed in dcc6b4d. Unix validation now requires the opened descriptor's `Stat_t.Uid` to equal the adapter process effective UID, with a focused mismatched-owner test. The full `make ci` suite passes.
